// node-vibrant/lib/test/builder.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var chai_1 = require("chai");
var Vibrant = require("../index");
// if (typeof Window !== undefined && this instanceof Window) {
//     Vibrant = this.Vibrant
// } else {
//     Vibrant = require('../index')
// }
//   Vibrant = require('../')
//   expect = require('chai').expect
describe('builder', function () {
    it('modifies Vibrant options', function () {
        var NOT_A_FILTER = function () { };
        var v = Vibrant.from('NOT_A_PATH')
            .maxColorCount(23)
            .quality(7)
            .useImageClass('NOT_AN_IMAGE')
            .useGenerator('NOT_A_GENERATOR')
            .useQuantizer('NOT_A_QUANTIZER')
            .clearFilters()
            .addFilter(NOT_A_FILTER)
            .build();
        var expected = {
            colorCount: 23,
            quality: 7,
            ImageClass: 'NOT_AN_IMAGE',
            quantizer: 'NOT_A_QUANTIZER',
            generator: 'NOT_A_GENERATOR',
            filters: [NOT_A_FILTER]
        };
        chai_1.expect(v.opts).to.deep.equal(expected);
    });
});
